### PR TITLE
Update tests due to the deprecated `qml.sample` usage when `shots=None`

### PR DESCRIPTION
**Pull request by antalszava**
Some test cases needed adjustment due to: https://github.com/PennyLaneAI/pennylane/pull/1522.

Note: since only the test cases are updated, there is no need to incorporate the changes in the release candidate branch. Having them in `master` will help with passing the checks in the plugin test matrix.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 flaky
 numpy
-pennylane>=0.15
+git+https://github.com/PennyLaneAI/pennylane.git@master
```
Comment (mlxd) [inline requirements.txt:3]: Will this be changed to `pennylane>=0.16` after the release?

**Review by albi3ro — APPROVED** [review]
seems all good.

File: requirements.txt (hunk 1 above)
Comment (antalszava) [reply requirements.txt:3]: The idea would be to merge the release candidate into `master` after the release. It seems, that there we're still pinning to `pennylane>=0.15`:
https://github.com/PennyLaneAI/pennylane-lightning/blob/937592796828ad131469a9258e715c7c7994c724/requirements.txt#L3

Having said that, let me undo this line, the change was made just to make sure that tests pass well with the `master` branch of PennyLane.

Comment (antalszava) [reply requirements.txt:3]:
```suggestion
pennylane>=0.15
```

**Pull request merged**